### PR TITLE
chore: upgrade closure-serializer and add more tests

### PR DESCRIPTION
**Pull request by thantos**
Uses the latest version of the closure serializer and adds more integration tests that output serialized and bundled handler code.

Use of Sfn in the closure has reduced from 1000s of lines to hundreds (less than 50k).

### CODE REVIEW

**Comment by sam-goodwin** [comment]
> less than 50k

Is that what it is now or used to be? Still seems high? What's the remaining bloat?

File: test/__snapshots__/serialize.test.ts.snap [hunk 1]
```diff
@@ -12380,6 +12380,1890 @@ function __f0() {
...
+                    return dynamo
+                        .putItem({
+                        ...rest,
+                        Item: Item,
+                        TableName: table.tableName,
+                    })
+                        .promise();
+                };
+
+    }
```
Comment (sam-goodwin) [inline test/__snapshots__/serialize.test.ts.snap:12585]: Does it pull in all of the table's native methods even if they aren't used?

File: test/__snapshots__/serialize.test.ts.snap [hunk 2]
```diff
@@ -12380,6 +12380,1890 @@ function __f0() {
...
+                    Entries: [
+                        {
+                            EventBusName: bus.eventBusName,
+                        },
+                    ],
+                });
+            };
+
+    }
+  }).apply(undefined, undefined).apply(this, arguments);
```
Comment (sam-goodwin) [inline test/__snapshots__/serialize.test.ts.snap:12723]: Does this have any performance impact on V8's ability to optimize this code?

File: test/__snapshots__/serialize.test.ts.snap [hunk 3]
```diff
@@ -12380,6 +12380,1890 @@ function __f0() {
 "
 `;
 
+exports[`serialize event bridge aws put events 1`] = `
```
Comment (sam-goodwin) [inline test/__snapshots__/serialize.test.ts.snap:12383]: Looks significantly less than 50k, but i am totally eye balling.

File: test/serialize.test.ts [hunk 4]
```diff
@@ -224,6 +232,104 @@ describe("serialize", () => {
...
+        () => () => {
+          return sfn({ input: {} });
+        },
+        []
+      );
+      expect(srlz).toMatchSnapshot();
+
+      const bundled = await bundle(srlz);
+      expect(bundled.text).toMatchSnapshot();
+      expect(bundled.text).toHaveLengthLessThan(BUNDLED_MAX_SIZE);
```
Comment (sam-goodwin) [inline test/serialize.test.ts:250]: Should we do this per test instead of capping each test at the same amount?

**Review by sam-goodwin — APPROVED** [review]

File: test/__snapshots__/serialize.test.ts.snap (hunk 2 above)
Comment (thantos) [reply test/__snapshots__/serialize.test.ts.snap:12723]: No idea, its what the closure serializer generates and have not played with it yet. Not 100% sure they need to wrap all of the function like this.

File: test/__snapshots__/serialize.test.ts.snap (hunk 3 above)
Comment (thantos) [reply test/__snapshots__/serialize.test.ts.snap:12383]: yeah, I didn't look, just didn't fail the test (50k).

File: test/serialize.test.ts (hunk 4 above)
Comment (thantos) [reply test/serialize.test.ts:250]: I set 50k as the max and then up them as needed. Could try lower and see what breaks.

File: test/__snapshots__/serialize.test.ts.snap (hunk 1 above)
Comment (thantos) [reply test/__snapshots__/serialize.test.ts.snap:12585]: Yes, originally I thought this was due to the namespace, but I changed to a constant and it still happened. Not sure why it cannot tree shake here.

File: test/__snapshots__/serialize.test.ts.snap (hunk 2 above)
Comment (sam-goodwin) [reply test/__snapshots__/serialize.test.ts.snap:12723]: Yeah I know we inherited this. Thought I would ask the question anyway. I'm curious of the answer.

File: test/__snapshots__/serialize.test.ts.snap (hunk 3 above)
Comment (sam-goodwin) [reply test/__snapshots__/serialize.test.ts.snap:12383]: Time to ratchet that down some more?

File: test/__snapshots__/serialize.test.ts.snap (hunk 2 above)
Comment (thantos) [reply test/__snapshots__/serialize.test.ts.snap:12723]: yeah, same, slowly making changes.

**Comment by thantos** [comment]
> > less than 50k
> 
> Is that what it is now or used to be? Still seems high? What's the remaining bloat?

looks like 10-15k is above average. with the tail being axios at 300k...

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**